src/lib/gadgets/lookup.unit-test.ts

```diff
@@ -0,0 +1,57 @@
+import { mod } from '../../bindings/crypto/finite_field.js';
+import { Field } from '../../lib/core.js';
+import { ZkProgram } from '../proof_system.js';
+import {
+  Spec,
+  boolean,
+  equivalentAsync,
+  fieldWithRng,
+} from '../testing/equivalent.js';
+import { Random } from '../testing/property.js';
+import { assert } from './common.js';
+import { Gadgets } from './gadgets.js';
+import { constraintSystem, contains } from '../testing/constraint-system.js';
+
+let maybeUint = (n: number | bigint): Spec<bigint, Field> => {
+  let uint = Random.bignat((1n << BigInt(n)) - 1n);
+  return fieldWithRng(
+    Random.map(Random.oneOf(uint, uint.invalid), (x) => mod(x, Field.ORDER))
+  );
+};
+
+let Lookup = ZkProgram({
+  name: 'lookup',
+  methods: {
+    three12Bit: {
+      privateInputs: [Field, Field, Field],
+      method(v0: Field, v1: Field, v2: Field) {
+        // Dummy range check to make sure the lookup table is initialized
+        // It should never fail because 64 > 12
+        Gadgets.rangeCheck64(v0);
+        Gadgets.three12Bit(v0, v1, v2);
+      },
+    },
+  },
+});
```

Comment:
I wasn't able to get the `ZkProgram` to compile unless I included `rangeCheck64`. Without it the process would just hang and not use any CPU time. I spent an hour or so trying to figure out why before I added `rangeCheck64`. Is this because of the [dummy gate thing](https://github.com/MinaProtocol/mina/blob/1d35680ad7aa654b57ba40562c6025d04ed89019/src/lib/crypto/kimchi_backend/gadgets/lookup.ml#L6)?

Reply:
Oh I see, so it was not fixed eventually? I would have sweared that it was addressed at some point in the summer... Well, yes, a few months ago, it wouldn't work unless your circuit had a previous gate which was already using the range check table, even if it is just a dummy zero value. But you needed to include that row... Seems like it is still the case 😞

Reply:
nice @jackryanservia - I guess you found out why https://github.com/o1-labs/o1js/pull/1253 wasn't working, so we can land that PR as well